RocketChat/Rocket.Chat.ReactNative
# fix: add checks to draft auto save

## Proposed changes

Currently the draft was being saved in ShareView and also when the user didn't edit anything they sent a message and clicked back.
This PR fixes the above errors.

## Issue(s)	


## How to test or reproduce

- With the app closed, type some text in shareExtension, wait 3 seconds, send the message, and return to the app, the text should not appear.
- With a saved draft, enter a room send the message that was saved in draft without clicking on the input component, send the message and quickly click return to leave the room, the draft must be empty.

## Screenshots

## Types of changes




## Checklist



## Further comments

Jira: [CORE-222]

[CORE-222]: https://rocketchat.atlassian.net/browse/CORE-222?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
